lib/xwax/pitch.h

```diff
@@ -1,71 +1,213 @@
+
 /*
- * Copyright (C) 2021 Mark Hills <mark@xwax.org>
- *
- * This file is part of "xwax".
- *
- * "xwax" is free software; you can redistribute it and/or modify it
- * under the terms of the GNU General Public License, version 3 as
- * published by the Free Software Foundation.
- *
- * "xwax" is distributed in the hope that it will be useful, but
- * WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
- * General Public License for more details.
+ * Kalman filter replacement for alpha-beta pitch estimator
+ * Model: constant velocity (x,v) with acceleration process noise
  *
- * You should have received a copy of the GNU General Public License
- * along with this program; if not, see <https://www.gnu.org/licenses/>.
+ * State:
+ *   x: displacement over the last interval (kept relative by subtracting dx at the end)
+ *   v: velocity (pitch)
  *
+ * Observation:
+ *   z = dx  (measured displacement over dt)
```

Comment:
For my understanding the Kalman filter has a model of the vinyl movements. All measurements are argumenting this model and the the output is read from the plausiblised model instead directly from the input. 
Is this correct? 

Can you add a brief description about these assumptions? Often if is obvious when writing the code but puzzling after years.

Reply:
This is a simple one-dimensional Kalman filter with a position-velocity model. It's the simplest filter that is out there. I can try to document the algorithm better